Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/ui/certificateselection/DccTicketingCertificateSelectionFragment.kt

```diff
@@ -1,24 +1,120 @@
 package de.rki.coronawarnapp.dccticketing.ui.certificateselection
 
 import android.os.Bundle
-import androidx.fragment.app.Fragment
-import android.view.LayoutInflater
 import android.view.View
-import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.lifecycleScope
 import androidx.navigation.fragment.navArgs
 import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.databinding.FragmentDccTicketingCertificateSelectionBinding
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.lists.decorations.TopBottomPaddingDecorator
+import de.rki.coronawarnapp.util.lists.diffutil.update
+import de.rki.coronawarnapp.util.ui.doNavigate
+import de.rki.coronawarnapp.util.ui.popBackStack
+import de.rki.coronawarnapp.util.ui.viewBinding
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModelsAssisted
+import setTextWithUrl
+import java.util.Locale
+import javax.inject.Inject
 
-class DccTicketingCertificateSelectionFragment : Fragment() {
+class DccTicketingCertificateSelectionFragment :
+    Fragment(R.layout.fragment_dcc_ticketing_certificate_selection), AutoInject {
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
 
     private val args by navArgs<DccTicketingCertificateSelectionFragmentArgs>()
+    private val binding: FragmentDccTicketingCertificateSelectionBinding by viewBinding()
+    private val viewModel: DccTicketingCertificateSelectionViewModel by cwaViewModelsAssisted(
+        factoryProducer = { viewModelFactory },
+        constructorCall = { factory, _ ->
+            factory as DccTicketingCertificateSelectionViewModel.Factory
+            factory.create(args.transactionContext)
```

Comment:
I know this was intended from the beginning to use `navArgs`, but let us use `DccTicketingSharedViewModel` as in `ConsentI`. this viewModel shares the updated `TransactionContext` in dcc_tickeing_flow.

Why?
TransactionContext data might have size of 10 MB or more (atm it can get so big, near future we have to set a limit up to 10 MB)

this amount of data in Bundle is way to big and exceeds the limit of how much data can passed using bundle see 
https://developer.android.com/guide/components/activities/parcelables-and-bundles#sdba
>When sending data via an intent, you should be careful to limit the data size to a few KB. Sending too much data can cause the system to throw a TransactionTooLargeException exception.